File: Unit2-03/KaiStack.ts
/**
 * This is the KaiStack class.
 *
 * author Kaitlyn Ip
 * version 1.0
 * since 2024-03-19
 */

class KaiStack {
    // array to store strings
    private strStack: string[] = []
    // constructor
    public getConstructor(): string {
        return this.constructor.name
    }

    // method to add a string to the stack
    push(input: string): void {
        this.strStack.push(input)
    }

    // method to remove a string from stack without using pop()
    pop(): string {
        if (this.strStack.length > 0) {
            const top_item = this.strStack[this.strStack.length - 1]
            this.strStack = this.strStack.slice(0, this.strStack.length - 1)
            return top_item
        } else {
            return "Stack is empty"
        }
    }

    // method to check if the stack is empty
    isEmpty(): boolean {
        if (this.strStack.length > 0) {
            return false
        } else {
            return true
        }
    }

    // method to say the size of the stack
    size(): number {
        const stackSize = this.strStack.length
        return stackSize
    }

    // method to show the stack
    show(): string {
        let values = ""
        for (let counter = 0; counter < this.strStack.length; counter++) {
                values += this.strStack[counter] + ", "
        }
        values = values.substring(0, values.length - 2)
        return values
    }
}

export default KaiStack
 

File: Unit2-02/java/KaiStack.java
/**
 * The KaiStack class.
 *
 * @author Kaitlyn Ip
 * @version 1.0
 * @since 2024-03-22
 */

final class KaiStack {

    /**
     * The main() method.
     *
     * @param args The command line arguments
     */
    private String[] stack;

    /**
     * Creates an empty stack.
     */
    final KaiStack() {
        stack = new String[0];
    }

    /**
     * Pushes an item onto the stack.
     *
     * @param item The item to push
     */
    public void push(String item) {
        final String[] newStack = new String[stack.length + 1];
        for (int counter = 0; counter < stack.length; counter++) {
            newStack[counter] = stack[counter];
        }
        newStack[newStack.length - 1] = item;
        stack = newStack;
    }

    /**
     * Pops the item out of the stack.
     */
        public String pop() {
            // Return top of the stack
        String returnValue = "";

        if (!stackAsList.isEmpty()) {
            String topItem = stackAsList.remove(stackAsList.size() - 1);
                returnValue = topItem;
        } else {
            returnValue = "The stack is empty!";
        }

	    return returnValue;
        }

    /**
     * Shows the stack.
     *
     * @return The stack
     */
    public String show() {
        String values = "";
        for (int counter = 0; counter < stack.length; counter++) {
            values += stack[counter];
            if (counter < stack.length - 1) {
                values += ", ";
            }
        }
        values += "";
        return values;
    }
}
